source/routing/raptor_api.cpp

```diff
@@ -1145,26 +1146,43 @@ pbnavitia::Response make_isochrone(RAPTOR &raptor,
     return pb_creator.get_response();
 }
 
+static void coord_to_string(std::stringstream& ss,
+                            const double& lon,
+                            const double& lat) {
+    ss << "[" << boost::lexical_cast<std::string>(lon) << ","
+       << boost::lexical_cast<std::string>(lat) << "]";
+}
+
 static void add_graphical_isochrone(const type::MultiPolygon& shape, PbCreator& pb_creator) {
-    auto pb_isochrone = pb_creator.add_graphical_isochrones();
-    auto pb_polys = pb_isochrone->mutable_geojson();
-    for (const auto& polygon: shape) {
-        auto p = pb_polys->add_polygons();
-        auto lo = p->mutable_outer();
-        for (const auto& coord: polygon.outer()) {
-            auto c = lo->add_coordinates();
-            c->set_lon(coord.lon());
-            c->set_lat(coord.lat());
+    std::stringstream geojson;
+    geojson << R"({"type":"MultiPolygon","coordinates":[)";
+    for (unsigned i = 0; i < shape.size(); i++) {
+        geojson << "[[";
+        for(unsigned j = 0; j < shape[i].outer().size(); j++) {
+            auto outer = shape[i].outer()[j];
+            coord_to_string(geojson, outer.lon(), outer.lat());
+            if (j == shape[i].outer().size() - 1) { continue; }
+            geojson << ",";
         }
```

Comment:
For this kind of pattern, I prefer doing like that:
```c++
auto it = shape[i].outer().begin();
const auto end = shape[i].outer().end();
if (it != end) { coord_to_string(geojson, it->lon(), it->lat()); ++it; }
for (; it != end; ++it) {
    coord_to_string(geojson << ",", it->lon(), it->lat());
}
```

Same for the main loop on shape.